### PR TITLE
Use RFC4180Parser as CSVParser

### LINKED ISSUE #6850: Allow the CSV parser to configure escaping
I have this test Excel file here:
https://static.imply.io/data/csv-test/EscapeTest.xlsx

It is pretty simple:

image

If you save it as CSV you get:

```
Value,Comment,Timestamp
3,"Lets do some ""normal"" quotes",2018-05-05T10:00:00Z
34,"Lets do some ""normal"", quotes with comma",2018-05-06T10:00:00Z
343,"Lets try \""it\"" with slash quotes",2018-05-07T10:00:00Z
545,"Lets try \""it\"", with slash quotes and comma",2018-05-08T10:00:00Z
65,Here I write \n slash n,2018-05-09T10:00:00Z
```

Clearly when Excel saves the file it does not care for the `\` escaping. Note that newlines in Excel get saved as literal new lines... so that is a lost cause.

This is an issue when trying to load it into Druid because Druid respects the `\` escapes. It would be great to have a way to specify the escaping scheme in the parser config. At least allow to opt out of the slash escapes.

In this case the data gets ingested as:

image

See how the slashes subtly messed things up.

Here is another dataset: https://static.imply.io/data/tweets/ira_small.csv

When parsing it text like: `"Как говорится: \""всё течет, всё изменяется\"". Украина как всегда обвиняет Россию в собственных проблемах. #ПровокацияКиева"`

Really messes up the time column:

image

### PR DESCRIPTION
**Pull request by SEKIRO-J**
Fixes #6850.

### Description

Current CSVParser  `opencsv.CSVParser` uses `\` as escape character while reading, and `"` while writing, and causing some mess.

I changed the `opencsv.CSVParser` to `opencsv.RFC4180Parser`,
which will ignore `\`, and use `"` as escaping character while reading.

#### Fixed the bug ...
For example:
`\"it\"` in Excel will be saved as `\""it\""` in CSV,
`\""it\""`  used to be read as `""it""`, now it is read as `\"it\"`

Note:
``\"it\"`` will be read as `\"it\"` as well

<hr>

This PR has:
- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/incubator-druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/incubator-druid/blob/master/licenses.yaml)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

<hr>

##### Key changed/added classes in this PR
 * `CSVParserTest`

### CODE REVIEW

**Comment by vogievetsky** [comment]
Yay! Thank you.

File: core/src/test/java/org/apache/druid/data/input/impl/CSVParserTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.data.input.impl;
```
Comment (surekhasaharan) [inline core/src/test/java/org/apache/druid/data/input/impl/CSVParserTest.java]: the test file should be in package `org.apache.druid.java.util.common.parsers` to follow the source file location.

File: core/src/test/java/org/apache/druid/data/input/impl/CSVParserTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        ImmutableMap.of("Value", "65", "Comment", "Here I write \\n slash n", "Timestamp", "2018-05-09T10:00:00Z")
+    );
+    final List<Map<String, Object>> parsedResult = new ArrayList<>();
+
+    for (String input : inputs) {
+      Map<String, Object> parsedLineList = parser.parseToMap(input);
+      parsedResult.add(parsedLineList);
+    }
+
+    Assert.assertTrue(expectedResult.equals(parsedResult));
```
Comment (surekhasaharan) [inline core/src/test/java/org/apache/druid/data/input/impl/CSVParserTest.java]: this works, but a better way to assert might be `Assert.assertEquals(expectedResult, parsedResult);`

File: core/src/test/java/org/apache/druid/data/input/impl/CSVParserTest.java (hunk 1 above)
Comment (SEKIRO-J) [reply core/src/test/java/org/apache/druid/data/input/impl/CSVParserTest.java]: right, will fix

**Comment by clintropolis** [comment]
I think this PR needs to have master branch merged in for it to pass CI

**Comment by surekhasaharan** [comment]
@SEKIRO-J could you please merge master, for teamcity and LGTM to pass

**Comment by capistrant** [comment]
I think a change like this should have more visibility in release notes in the future. This has the ability to break ingestion jobs that worked in previous versions of druid if they relied on the legacy parser. Any way to retroactively make updates to the druid 17 release notes to call out the new parser coming into play? I'm sure it would save someone the pain that my team felt when our clients started seeing their ingestion jobs start failing after our recent upgrade from druid 15 to druid 18.

**Pull request merged**